### PR TITLE
Brackets and bezier lines

### PR DESCRIPTION
**Pull request by jkrumbiegel**
- allow cairomakie to draw bezierpath lines, add convert to vector of points for bp
- add a bracket recipe

image

# Description

Fixes # (issue)

Add a description of your PR here.

## Type of change

Delete options that do not apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] Added an entry in NEWS.md (for new features and breaking changes)
- [x] Added or changed relevant sections in the documentation
- [ ] Added unit tests for new algorithms, conversion methods, etc.
- [ ] Added reference image tests for new plotting functions, recipes, visual options, etc.

### CODE REVIEW

**Comment by asinghvi17** [comment]
Looks pretty good to me - can we merge this?

**Comment by jkrumbiegel** [comment]
I think so, the only remaining question was what to do with the conversion method for BezierPath. I'm converting to a line essentially, so that glmakie can fall back to that. But that means you can't use `poly` with a BezierPath. Maybe we should just tackle that later though and just keep the lines for now.

**Comment by asinghvi17** [comment]
We could override the `convert_arguments` for Poly specifically, if the only adjustments needing to be made are that the path should be closed.  IMO, that could be tackled later fairly easily.

PS: this will help out in a lot of other places, since you could do e.g. bars, arrows, etc with different styles.

File: src/basic_recipes/bracket.jl [hunk 1]
```diff
@@ -0,0 +1,146 @@
+"""
+    bracket(x1, y1, x2, y2; kwargs...)
+    bracket(x1s, y1s, x2s, y2s; kwargs...)
+    bracket(point1, point2; kwargs...)
+    bracket(vec_of_point_tuples; kwargs...)
+
+Draws a bracket between each pair of points (x1, y1) and (x2, y2) with a text label at the midpoint.
+
+By default each label is rotated parallel to the line between the bracket points.
+
```
Comment (asinghvi17) [inline src/basic_recipes/bracket.jl:10]:
```suggestion

In order to implement a new style for brackets, overload `Makie.bracket_bezierpath(::Val{:mybracketstyle}, p1, p2, d, width)`, 
which should return a [`BezierPath`](@ref).
```

Comment (jkrumbiegel) [reply src/basic_recipes/bracket.jl:10]: I don't want to make that API, yet, in case I find there's more to it than I currently have.

**Pull request merged**